lib/associations/belongs-to-many.js

```diff
@@ -298,6 +298,46 @@ var BelongsToMany = function(source, target, options) {
 
 util.inherits(BelongsToMany, Association);
 
+BelongsToMany.prototype.get = function(instances, options) {
+  var throughModel = this.throughModel;
+
+  var throughOptions = _.assign({
+    where: _.set({}, this.foreignKey, {$in: _.pluck(instances, this.source.primaryKeyAttribute)}),
+    logging: options.logging
+  });
+
+  var association = this;
+
+  return throughModel.findAll(throughOptions).then(function (throughs) {
+    var otherKey = association.otherKey
+      , targetIds = _.chain(throughs).pluck(otherKey).uniq().value();
+
+    var targetOptions = _.chain({
+      where: _.set({}, association.target.primaryKeyAttribute, {$in: targetIds})
+    })
+        .assign(association.target.$optClone(options))
+        .omit(['include'])
+        .value();
+
+    return Utils.Promise.join(throughs, _.isEmpty(targetIds) ? null : association.target.findAll(targetOptions));
```

Comment:
Sequelize doesn't currently support duplicate entries in the join model for a source and target, although that has been a feature request before with better unique: false support.

Ideally for a belongsToMany association with `unique: true` (or not `unique: false`) rather we should use an optimized single query.

It should be possible to do an INNER JOIN and a UNION ALL across a set of selects with inner joins.

/cc @janmeier